code/codegen/src/main/kotlin/com/android/gradle/replicator/resgen/DrawableResourceGenerator.kt

```diff
@@ -44,32 +42,48 @@ class DrawableResourceGenerator (
     )
 
     override fun generateResource(
-            number: Int,
-            outputFolder: File,
-            resourceQualifiers: List<String>,
-            resourceExtension: String
+            properties: AbstractAndroidResourceProperties,
+            outputFolder: File
     ) {
-        repeat(number) {
+        // Sanity check
+        if (properties.propertyType != ResourcePropertyType.DEFAULT) {
```

Comment:
I see this check everywhere but I am not very clear when it can happen, can you elaborate the //sanity check comment to include more information why this is possible outcome.

Reply:
This is supposed to catch bugs when reading the metadata file. It should not happen under normal circumstances. I will add a comment to it.

Reply:
For instance, if the metadata reader creates a DefaultResourceProperty instead of a ValuesResourceProperty, I could get some cryptic errors down the line.